diff --git a/node_modules/http-call/lib/http.js b/node_modules/http-call/lib/http.js
index dbef094..8f7591c 100644
--- a/node_modules/http-call/lib/http.js
+++ b/node_modules/http-call/lib/http.js
@@ -326,7 +326,11 @@ class HTTP {
                 });
             }
             this.request.on('error', reject);
-            this.request.on('timeout', reject);
+            this.request.on('timeout', () => {
+                const error = new Error('Connection timeout');
+                error.code = 'ETIMEDOUT';
+                reject(error);
+            });
             if (this.options.body && deps_1.deps.isStream.readable(this.options.body)) {
                 this.options.body.pipe(this.request);
             }
